src/components/timeline/TimelineMarkerCanvas.js

```diff
@@ -146,47 +200,27 @@ class TimelineMarkerCanvas extends PureComponent {
 
           const x: CssPixels =
             (startTime - viewportLeft) * containerWidth / viewportLength;
-          const y: CssPixels = rowIndex * ROW_HEIGHT - viewportTop;
-          const w: CssPixels = Math.max(
-            10,
-            (endTime - startTime) * containerWidth / viewportLength
-          );
-          const h: CssPixels = ROW_HEIGHT - 1;
+          const y: CssPixels = rowIndex * rowHeight - viewportTop;
+          const w: CssPixels =
+            (endTime - startTime) * containerWidth / viewportLength;
+          const h: CssPixels = rowHeight - 1;
+          const text = markerTiming.label[i];
 
           const tracingMarkerIndex = markerTiming.index[i];
           const isHovered = hoveredItem === tracingMarkerIndex;
-          ctx.fillStyle = isHovered ? 'Highlight' : '#8296cb';
-
-          if (w >= h) {
-            this.drawRoundedRect(ctx, x, y + 1, w, h - 1, 1);
-
-            const text = markerTiming.label[i];
-            // Draw the text label
-            // TODO - L10N RTL.
-            // Constrain the x coordinate to the leftmost area.
-            const x2: CssPixels = Math.max(x, 0) + TEXT_OFFSET_START;
-            const w2: CssPixels = Math.max(0, w - (x2 - x));
-
-            if (w2 > textMeasurement.minWidth) {
-              const fittedText = textMeasurement.getFittedText(text, w2);
-              if (fittedText) {
-                ctx.fillStyle = isHovered ? 'HighlightText' : '#ffffff';
-                ctx.fillText(fittedText, x2, y + TEXT_OFFSET_TOP);
-              }
-            }
+          if (isHovered) {
+            hoveredElement = { x, y, w, h, text };
```

Comment:
magic: defering the drawing of this marker to the end, so that it's well visible :)